### ISSUE TITLE
Does fail on errors

### ISSUE THREAD

**Issue by GPetrites** [VisualBean/VSTSTools#3]
If I don't have security configured correctly for build service account, I can see in the logs that the error occurred, but the task completes with a success.

**Comment by VisualBean** [comment]
I will look into this and make sure it fails when it should :) - thanks

**Comment by VisualBean** [comment]
unfortunately, no exception is thrown - so a "simple" try catch will not alleviate this problem. 
this would also cause the task to fail anyway.

I will get in touch with microsoft to see if this can be resolved in their end :)

Seems like an implementation issue if the command does not throw an exception, but gives an error.

**Comment by bqstony** [comment]
Hi

Do you have any workaround for this issue? I have the same behavior.

log output:
```
2017-07-19T11:21:04.1349417Z ##[command] git checkout buildstorage
2017-07-19T11:21:04.1969479Z error: pathspec 'buildstorage' did not match any file(s) known to git.
2017-07-19T11:21:04.2049487Z ##[command] git tag  Release/identity/1.0.0-alpha9
2017-07-19T11:21:04.2669549Z fatal: tag 'Release/identity/1.0.0-alpha9' already exists
2017-07-19T11:21:04.2729555Z ##[command] git push origin Release/identity/1.0.0-alpha9
2017-07-19T11:21:05.0630345Z Everything up-to-date
```

**Comment by VisualBean** [comment]
@bqstony.
You issue doesnt seem to be security. The log will say something about missing permissions.

You seem to have not checked the "force" checkbox.
As the tag already exists.
But does the task not give an error? - I am still looking into the error handling though.

**Comment by VisualBean** [comment]
I might have a fix for the issue. Will try to push changes later today.

**Comment by VisualBean** [comment]
Fixed in 6bfd4b1c10077035ba1d3d5ff0710e67cf60f851
